### DISCUSSION TITLE
IPSEC Tunnel between Strongswan - Cisco 7201 suddenly not working

### DISCUSSION THREAD

**Discussion by XrayKilo** [strongswan/strongswan#619] [Q&A]
Hello,
We have a Ubuntu 20.04.2 VPS with strongswan 5.8.2-1ubuntu3.1 installed, we use it to create an IPSEC IKEv1 tunnel to a Cisco 7201 managed by our telecom partner, to access a private VPN. It worked ok for a year, then in August it stopped and got stuck in "Connecting" status on the server side, reboots and restarts on the server side and on the router did not help. After about 48 hours it started working again. A day ago it stopped again, a server reboot, “clear crypto session” and “clear crypto sa peer” on the router did not help. There were no changes on either side, we can't get this tunnel to work, and obviously important data doesn't get sent.
I am attaching the configs and logs, as supplied by the telco, and collected from our server.
Server_IP is the Ubuntu VPS, Endpoint_IP is the Cisco, and the VPN_Subnet is the subnet accessed by the server on the telco side.
The only unusual thing I see is that the strongswan inserts a route:
VPN_Subnet/24 via Default_Gateway dev eth0 table 220 proto static src Server_IP ,where Default_Gateway is the default network gateway of the eth0.
No one has any idea how to fix this, so please help.
[strongswan_log.txt](https://github.com/strongswan/strongswan/files/7224229/strongswan_log.txt)
[cisco_config.txt](https://github.com/strongswan/strongswan/files/7224230/cisco_config.txt)
[ip_address.txt](https://github.com/strongswan/strongswan/files/7224231/ip_address.txt)
[iptables_save.txt](https://github.com/strongswan/strongswan/files/7224232/iptables_save.txt)
[cisco_log.txt](https://github.com/strongswan/strongswan/files/7224233/cisco_log.txt)
[ipsec_conf.txt](https://github.com/strongswan/strongswan/files/7224234/ipsec_conf.txt)
[ip_routes.txt](https://github.com/strongswan/strongswan/files/7224235/ip_routes.txt)
[ipsec_statusall.txt](https://github.com/strongswan/strongswan/files/7224236/ipsec_statusall.txt)

**Comment by tobiasbrunner** [comment]
Make sure only a single instance of `charon` (and/or `charon-systemd` or any other process listening on UDP ports 500/4500) is running. Other than that, it just looks like a network/routing/firewall issue or a problem on the other peer itself, as strongSwan, besides sending its own requests, does seem to receive requests from the peer to which it replies, but we don't see follow-ups to either (i.e. packets from strongSwan to the other box might get dropped/rerouted somewhere).

**Reply by XrayKilo** [reply]
Thank you. We opened ports 500 and 4500 on the server, the log changed a bit on our side, still doesn't work.
Our partners indicate that the PFS is set to off, and the security association lifetime is 4608000 kilobytes/3600 seconds, do I understand correctly that esp=aes256-sha1! is appropriate, and that ikelifetime should be modified to 1h?

.

**Reply by tobiasbrunner** [reply]
>  the log changed a bit on our side, still doesn't work.

How? Looks pretty much the same to me.

But something with the timing/clock on your host is weird. You restarted the daemon and the wall clock time in the log is 14:10:02  for all the messages, however, we see multiple retransmits (which are usually [scheduled to spread over several minutes](https://wiki.strongswan.org/projects/strongswan/wiki/Retransmission)) and a half-open SA timeout (which is scheduled after 30 seconds, by default). So either your config is broken or the monotonic clock (strongSwan uses `clock_gettime(CLOCK_MONOTONIC, ...)` to schedule events) runs fast on your host.

**Reply by XrayKilo** [reply]
Indeed, I copied the log from the syslog, and sometimes the retransmits are properly spaced, and sometimes they are written all at the same time. But I also notice that when ipsec writes them then it's all at the same time, and when charon does it it's properly spaced.
I launch strongswan using systmectl start ipsec, and when I stop it I see no service listening on port 500, so it's shutting down properly, but then why do I have both ipsec and charon?
[systemctl.txt](https://github.com/strongswan/strongswan/files/7225436/systemctl.txt)
[syslog_time.txt](https://github.com/strongswan/strongswan/files/7225440/syslog_time.txt)

**Reply by tobiasbrunner** [reply]
Probably depends on your logging settings (see [LoggerConfiguration](https://wiki.strongswan.org/projects/strongswan/wiki/LoggerConfiguration)). The _strongswan-starter_ systemd service unit (_ipsec_ is an alias to it apparently) starts the _starter_ daemon with `--nofork` so there will be output via stdout/stderr that's logged directly (under _ipsec_). If _charon_ also logs via syslog, you get each message twice. Possible that systemd uses some buffering and bunches the messages together (still weird, though).

So if the timing isn't the problem, you still have that networking issue from this host to the Cisco box.

**Comment by XrayKilo** [comment]
Thanks. Any suggestions on how to debug the networking for the ipsec, so I can point the finger in the right direction? 
The pings are going in both directions, both the partner telco and the hosting provider insist nothing changed on their sides and that there are no firewalls between us, turning off my firewall didn't help. 
Also, I copied the secrets and ipsec.conf files to another server, the partner changed the IPs and the ACL permissions on their side, and everything works just fine.

**Reply by Thermi** [reply]
I'd verify the timing of the retransmits via tcpdump and make sure you have a file log with millisecond timestamps. Then you can rule out it's some weirdness going on. I'd also verify if ntp is working correctly.

In parallel, while those tests are run and data is captured, verify you have stateless rules for the IPsec related ports and protocols (esp, ah, UDP port 500/4500) (then you can rule out timing weirdness related to local stateful firewall record timeouts).

Then we need to continue looking at the exact results.

**Reply by XrayKilo** [reply]
Ok, before we got to do all these we tried to test the tunnel on another cloud server, by changing the endpoint address on the cisco router, it functioned as expected with the same strongswan configuration file. While this change was occuring, I purged the strongswan package again, and tried to install libreswan instead, which showed some strange systemctl errors on launch, so I purged it too and installed strongswan back. About this time the cisco side changed the IP's and the ACLs back to our server, and everything started working again.
1. Maybe installing and purging libreswan affected the system in some way a simple purge+install of strongswan didn't.
2. Maybe changing the endpoint IP and the ACL IP on the Cisco provoked a deeper reset than a simple “clear crypto session” and “clear crypto sa peer”, also their IOS is 6 years old (15.0(1)M5)
3. Maybe something changed in the network between us, despite the traceroute remaining the same.
Anyway, it's working now, so thank you for your time and the suggestions.

**Reply by Thermi** [reply]
>  and tried to install libreswan instead, 

The configs are not compatible to each other, although the syntax of the old config file looks about the same.

> Maybe installing and purging libreswan affected the system in some way a simple purge+install of strongswan didn't.

No. Except maybe if you inadvertently tried running both IKE daemons at the same time. That could lead to trouble.

> Maybe changing the endpoint IP and the ACL IP on the Cisco provoked a deeper reset than a simple “clear crypto session” and “clear crypto sa peer”, also their IOS is 6 years old (15.0(1)M5)

Sometimes there are weird issues with PSK authentication and certain IP addresses. I haven't yet had the opportunity to figure out if it's just a problem in the other peer's implementation (chooses wrong PSK, screws up encoding/processing, ...) or a problem in strongSwan. Up to now that problem could always be solved by changing the public address of the strongSwan endpoint. So sometimes things are weird.

> Maybe something changed in the network between us, despite the traceroute remaining the same.
Anyway, it's working now, so thank you for your time and the suggestions.

Can be. Shouldn't though.

Yw.
